### PR TITLE
Render User Data on Table form API

### PR DESCRIPTION
**Pull request by DKnehi**

### CODE REVIEW

File: javascript-practice/src/js/controllers/dashboardController.js [hunk 1]
```diff
@@ -45,4 +49,19 @@ export default class DashboardController {
       throw new Error('Failed to add user.');
     }
   };
+
+  renderTable = async () => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/dashboardController.js]: Should clarify what table?

File: javascript-practice/src/js/controllers/dashboardController.js [hunk 2]
```diff
@@ -45,4 +49,19 @@ export default class DashboardController {
...
+    try {
+      const data = await this.service.getAllUser(); 
+      const tableHTML = generateTableHTML(data);
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/controllers/dashboardController.js]: Why controller handle html code?

File: javascript-practice/src/js/services/userService.js [hunk 3]
```diff
@@ -42,4 +42,18 @@ export default class UserService {
       throw new Error('Failed to add user.');
     }
   };
+
+//   fetchDataFromMockAPI = async () => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/services/userService.js]: Remove if not use

File: javascript-practice/src/js/templates/userTemplate.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import editIcon from '../../assets/icons/arrow-circle-down.svg';
+import deleteIcon from '../../assets/icons/arrow-circle-down-copy.svg';
+
+const generateTableRowHTML = (item) => {
+  let roleClass = '';
+
+  const role = item.role.trim().toLowerCase();
+
+  if (['supper admin', 'admin', 'hr admin'].includes(role)) {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js]: Should be define in constant

File: javascript-practice/src/js/templates/userTemplate.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import editIcon from '../../assets/icons/arrow-circle-down.svg';
+import deleteIcon from '../../assets/icons/arrow-circle-down-copy.svg';
+
+const generateTableRowHTML = (item) => {
+  let roleClass = '';
+
+  const role = item.role.trim().toLowerCase();
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js]: Why trim?

File: javascript-practice/src/js/templates/userTemplate.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  if (['supper admin', 'admin', 'hr admin'].includes(role)) {
+    roleClass = 'table-primary-role';
+  } else {
+    roleClass = 'table-secondary-role';
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js]: Do we really need table as a part of class name?

File: javascript-practice/src/js/templates/userTemplate.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  const addedDate = new Date(item.addedDate);
+  const day = addedDate.getDate().toString().padStart(2, '0');
+  const monthIndex = addedDate.getMonth();
+  const monthNames = [
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js]: Constant

File: javascript-practice/src/js/templates/userTemplate.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+
+  const month = monthNames[monthIndex];
+  const year = addedDate.getFullYear().toString();
+  const formattedDate = `${day} ${month}, ${year}`;
+
+  return `
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js:37]: Indent of this block is incorrect

File: javascript-practice/src/js/templates/userTemplate.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  <tr class="table-row">
+  <td class="table-data table-cell">
+    <p class="table-data-name">${item.userName}</p>
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/templates/userTemplate.js]: Please remove all classes that you not use for styling

File: javascript-practice/src/styles/sections/list-users.css [hunk 10]
```diff
@@ -14,10 +14,20 @@
...
+
 .table-data-date {
   line-height: 15px;
 }
 
 .list-user-table-icon {
   margin-right: 20px;
 }
+
+.data-icon {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/sections/list-users.css]: All image use div for sizing, then all icon/image should get this style. So please use it as common class

File: javascript-practice/src/js/controllers/dashboardController.js (hunk 1 above)
Comment (DKnehi) [reply javascript-practice/src/js/controllers/dashboardController.js]: i done fix

File: javascript-practice/src/js/views/dashboardView.js [hunk 11]
```diff
@@ -208,6 +212,19 @@ export default class DashboardView {
     showToast(message);
   };
 
+  renderTableListUsers = (data) => {
+    const tableHTML = generateTableHTML(data);
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/dashboardView.js:216]: Should have empty before const

**Pull request merged**